edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/EdgeHubFixtureCollection.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.E2E.Test
+{
+    using System.Collections.Generic;
+
+    public static class EdgeHubFixtureCollection
+    {
+        static readonly object UpdateLock = new object();
+        static readonly IList<ProtocolHeadFixture> Fixtures = new List<ProtocolHeadFixture>();
+
+        internal static ProtocolHeadFixture GetFixture()
+        {
+            lock (UpdateLock)
+            {
+                // Before creating a new fixture, we need to ensure that other ProtocolHeadFixture
+                // instances are closed. This is because a ProtocolHeadFixture encapsulates a
+                // socket connection over an address and only one usage of this address is allowed
+                // at a time.
+                foreach (ProtocolHeadFixture fixture in Fixtures)
+                {
+                    if (!fixture.IsClosed)
+                    {
+                        fixture.CloseAsync().Wait();
```

Comment:
Is there a reason we maintain this list? Why not just have a current Protocol Head?
Also, what if 2 tests ask for this at the same time?

Reply:
Thinking about it, you're right, we can just close the current ProtocolHeadFixture and create a new one as current instead. I'll try it out.

Yeah, parallel tests won't work. Currently, all integration tests running sequentially.

The reason all these changes were needed to the ProtocolHeadFixture (and the new code added in #1759 ) was to have a way to close the DbStoreProvider which in turn will trigger a backup. To test restores, I had to recreate the EdgeHub app (running in the same test process) so that the in-memory store gets re-created and we can test whether restore worked.

The problem I was facing with simply disposing ProtocolHeadFixture and recreating a new one was that:
1. You can't have multiple ProtocolHeadFixtures due to the following reason - a ProtocolHeadFixture encapsulates a socket connection over an address and only one usage of this address is allowed at a time.
2. If you however try to only recreate a single ProtocolHeadFixture after disposing the existing one, I was getting errors because somehow disposing the fixture also leads to the disposal of the singleton Logger instance (created by Logger.Factory) which causes any attempts at recreation to fail.